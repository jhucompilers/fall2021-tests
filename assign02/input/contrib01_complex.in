function foo(currVal, base) {
    var ret;
    ret = -1;

    if (currVal <= 2) {
        if (currVal == 2) {ret = 1;}
        else {ret = 0;}
    }

    if (ret == -1) {
        //get remainder
        var tempVal;
        tempVal = currVal;
        while(tempVal >= base){
            tempVal = tempVal - base;
        }
        if (tempVal == 0) {ret = 0;}
    }

    if (ret == -1) {
        if (base * base > currVal) {ret = 1;}
    }

    if (ret == -1) {
        ret = foo(currVal, base+1);
    }

    ret;
}

function fooSquare(o, p) {
    var up;
    up = -1;
    if((p * p >= o) && (p * p == o)) {
        up = 1;
    }
    if((p * p >= o) && (p * p != o)) {
            up = 0;
    }
    if (up == -1) {
        up = fooSquare(o, p + 1);
    }
    up;
}





// LOOP HANDLING FOR INTEGERS 1-49
var i, n;
i = 1;
n = 49;


function inc() {
    i = i+1;
}

function printnumspace(num) {
    var k;
    k = 0;
    while (k < num) {
        printspace();
        k = k +1;
    }
}

var ps;
ps = printnumspace;

while (i < n) {
    print(i);
    if (i < 10) {ps(5);
    } else {ps(4);}

    print(foo(i, 2)); //check if prime
    ps(2);
    println(fooSquare(i, 0)); //check if perfect square

    inc();
}




