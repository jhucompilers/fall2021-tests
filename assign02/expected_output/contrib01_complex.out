1     0  1
2     1  0
3     1  0
4     0  1
5     1  0
6     0  0
7     1  0
8     0  0
9     0  1
10    0  0
11    1  0
12    0  0
13    1  0
14    0  0
15    0  0
16    0  1
17    1  0
18    0  0
19    1  0
20    0  0
21    0  0
22    0  0
23    1  0
24    0  0
25    0  1
26    0  0
27    0  0
28    0  0
29    1  0
30    0  0
31    1  0
32    0  0
33    0  0
34    0  0
35    0  0
36    0  1
37    1  0
38    0  0
39    0  0
40    0  0
41    1  0
42    0  0
43    1  0
44    0  0
45    0  0
46    0  0
47    1  0
48    0  0
Result: <void>
